spec/models/course_spec.rb

```diff
@@ -142,9 +142,457 @@
     end
   end
 
-  describe '#upload_assignment_list' # TODO
-  describe '#get_required_files' # TODO
+  describe '#upload_assignment_list' do
+    let!(:test_course) do
```

Comment:
Two comments:

1. In general to create model instances, you should use the factory version (`create(:course)`), which will populate attributes for you.
2. For this specifically, you shouldn't define `test_course` at all, but instead use the `course` variable that's been defined in a `let` at the top of this file.